pkg/interfaces/contracts/pool-utils/IManagedPool.sol

```diff
@@ -0,0 +1,316 @@
+     * @return endTime - The timestamp when the weight update will end.
+     * @return startWeights - The starting weights, when the weight change was initiated.
+     * @return endWeights - The final weights, when the current timestamp >= endTime.
+     */
+    function getGradualWeightUpdateParams()
+        external
+        view
+        returns (
+            uint256 startTime,
+            uint256 endTime,
+            uint256[] memory startWeights,
+            uint256[] memory endWeights
+        );
+
+    // Swap enable/disable
+
+    /**
+     * @notice Enable or disable trading.
+     * @dev Emits the SwapEnabledSet event. This is a permissioned function.
+     * @param swapEnabled - The new value of the swap enabled flag.
+     */
+    function setSwapEnabled(bool swapEnabled) external;
+
+    /**
+     * @notice Returns whether swaps are enabled.
+     */
+    function getSwapEnabled() external view returns (bool);
+
+    // LP Allowlist
+
+    /**
+     * @notice Enable or disable the LP allowlist.
+     * @dev Note that any addresses added to the allowlist will be retained if the allowlist is toggled off and
+     * back on again, because this action does not affect the list of LP addresses.
+     * Emits the MustAllowlistLPsSet event. This is a permissioned function.
+     * @param mustAllowlistLPs - The new value of the mustAllowlistLPs flag.
+     */
+
+    function setMustAllowlistLPs(bool mustAllowlistLPs) external;
+
+    /**
+     * @notice Adds an address to the LP allowlist.
+     * @dev Will fail if the address is already allowlisted.
+     * Emits the AllowlistAddressAdded event. This is a permissioned function.
+     * @param member - The address to be added to the allowlist.
+     */
+    function addAllowedAddress(address member) external;
+
+    /**
+     * @notice Removes an address from the LP allowlist.
+     * @dev Will fail if the address was not previously allowlisted.
+     * Emits the AllowlistAddressRemoved event. This is a permissioned function.
+     * @param member - The address to be removed from the allowlist.
+     */
+
+    function removeAllowedAddress(address member) external;
+
+    /**
+     * @notice Returns whether the allowlist for LPs is enabled.
+     */
+    function getMustAllowlistLPs() external view returns (bool);
+
+    /**
+     * @notice Check whether an LP address is on the allowlist.
+     * @dev This simply checks the list, regardless of whether the allowlist feature is enabled.
+     * @param member - The address to check against the allowlist.
+     * @return true if the given address is on the allowlist.
+     */
+    function isAddressOnAllowlist(address member) external view returns (bool);
+
+    // Management fees
+
+    /**
+     * @notice Collect any accrued AUM fees and send them to the pool manager.
+     * @dev This can be called by anyone to collect accrued AUM fees - and will be called automatically on
+     * joins and exits.
```

Comment:
```suggestion
     * @dev This can be called by anyone to collect accrued AUM fees - and will be called automatically
     * whenever the supply changes (e.g., joins and exits, add and remove token), and before the fee
     * percentage is changed by the manager, to prevent fees from being applied retroactively.
```